chia/wallet/wallet_blockchain.py

```diff
@@ -193,20 +193,24 @@ def get_latest_timestamp(self) -> uint64:
     def contains_block(self, header_hash: bytes32) -> bool:
         return header_hash in self._block_records
 
+    async def contains_block_from_db(self, header_hash: bytes32) -> bool:
+        return header_hash in self._block_records
+
     def contains_height(self, height: uint32) -> bool:
         return height in self._height_to_hash
 
     def height_to_hash(self, height: uint32) -> bytes32:
         return self._height_to_hash[height]
 
     def try_block_record(self, header_hash: bytes32) -> Optional[BlockRecord]:
-        if self.contains_block(header_hash):
-            return self.block_record(header_hash)
-        return None
+        return self._block_records.get(header_hash)
 
     def block_record(self, header_hash: bytes32) -> BlockRecord:
         return self._block_records[header_hash]
 
+    async def get_block_record_from_db(self, header_hash: bytes32) -> Optional[BlockRecord]:
+        return self._block_records.get(header_hash)
+
```

Comment:
not sure i get the naming, where is the db access ?

Reply:
I'll add a comment. This is implementing the `BlockchainInterface` which is kind-of like an abstract base class